Source/Core/Core/IPC_HLE/WII_IPC_HLE_Device.h

```diff
@@ -41,22 +40,89 @@ enum IOSReturnCode : s32
   IPC_EESEXHAUSTED = -1016,  // Max of 2 ES handles exceeded
 };
 
-// A struct for IOS ioctlv calls
-struct SIOCtlVBuffer
+struct IOSRequest
 {
-  explicit SIOCtlVBuffer(u32 address);
-
-  const u32 m_Address;
-  u32 Parameter;
-  u32 NumberInBuffer;
-  u32 NumberPayloadBuffer;
-  u32 BufferVector;
-  struct SBuffer
+  u32 address = 0;
+  IPCCommandType command = IPC_CMD_OPEN;
+  u32 fd = 0;
+  explicit IOSRequest(u32 address);
+  virtual ~IOSRequest() = default;
+  void SetReturnValue(s32 new_return_value) const;
```

Comment:
It feels a little weird that the _request_ has a/gets the return value, and then a _reply_ is sent (which apparently does not know about the actual value).
I do see why it is easier to put it here (mostly because it has the address and everything), but IMO it logically belongs more to `GetDefaultReply`, `GetNoReply` and similar; or least somewhere in that area.
What do you think about handling this inside `WII_IPC_HLE_Interface::HandleCommand` (unless I missed other places where those IOS functions are called from) by actually returning the return value as part of the reply and doing the result write there (or not, if there is no result to be written)?

Or did I forget about something important here which makes this more complicated than that?